File: 그리디/greedy_2.py
n, m, k = map(int, input().split())
num_data = list(map(int, input().split()))

num_data.sort()
first = num_data[n - 1]
second = num_data[n - 2]

count = int(m / (k + 1)) * k
count += m % (k + 1)

result = 0
result += (count) * first
result += (m - count) * second

print(result)

File: 그리디/greedy_2.js
const solution = (n, m, k, arr) => {
    arr.sort((a, b) => b - a);
    let first = arr[0];
    let second = arr[1];

    let result = 0;

    let count = Math.floor(m / k) * k;
    result += count * first;
    result += (m - count) * second;

    return result;
}

let arr = [2, 4, 5, 4, 6];
console.log(solution(5, 8, 3, arr));